crates/nu-protocol/src/value/mod.rs

```diff
@@ -1013,7 +1003,7 @@ impl Value {
                                     x.0 == column_name
                                 }
                             }) {
-                                current = found.1.clone();
+                                current = found.1.clone(); // TODO: avoid clone here
```

Comment:
Is `.into_iter()` not an option here?

The existing code is still defensive around finding more than one entry with the same column name. in the future we can probably replace that all with `Record::remove()`

**Edit:** As the `upsert`/`update` logic makes the assumptions that the first hit is the true hit, you could go for that and we see what starts to burn 😆

Reply:
Unfortunately, the records columns are used below in `did_you_mean(val.columns(), column_name)`, so we can't clone here.

Reply:
Ah then an `if let Some(val) = record.remove(col)` magic could be the workaround, right?

Reply:
There's a predicate for the case insensitivity. So, maybe `index_of` followed by `swap_remove`?

Reply:
Grr sloppy `insensitive`. In my view that should have no business in any cell-path access but I digress.

Yeah sounds good. Adding `remove_index`/`swap_remove_index` to record should be trivial.